src/Microsoft.AspNetCore.Mvc.Razor/Internal/RazorViewEngineOptionsSetup.cs

```diff
@@ -4,16 +4,14 @@
 using System;
 using System.Linq;
 using Microsoft.AspNetCore.Hosting;
-using Microsoft.AspNetCore.Mvc.Razor;
 using Microsoft.CodeAnalysis;
 using Microsoft.Extensions.Options;
 
-namespace Microsoft.AspNetCore.Mvc
+namespace Microsoft.AspNetCore.Mvc.Razor.Internal
 {
     /// <summary>
     /// Sets up default options for <see cref="RazorViewEngineOptions"/>.
     /// </summary>
-    [Obsolete("This type is for internal use and will be removed in a future version.")]
     public class RazorViewEngineOptionsSetup : ConfigureOptions<RazorViewEngineOptions>
```

Comment:
Could you also change this type to derive from `IConfigureOptions<RazorViewEngineOptions>`. Look at the other OptionsSetup type to see what they do.